### PR TITLE
[3.10] gh-89301: Fix regression with bound values in traced SQLite statements

### PR DESCRIPTION
**Pull request by erlend-aasland**
(cherry picked from commit 721aa96540bb96700f8c4bab0b4095b43491dca1)

### CODE REVIEW

File: Lib/sqlite3/test/hooks.py [hunk 1]
```diff
@@ -261,6 +272,21 @@ def trace(statement):
         cur.execute(queries[1])
         self.assertEqual(traced_statements, queries)
 
+    def test_trace_expanded_sql(self):
```
Comment (JelleZijlstra) [inline Lib/sqlite3/test/hooks.py:275]: Why aren't the tests from `test_trace_too_much_expanded_sql` included? Does the framework for those not exist in 3.10?

Comment (erlend-aasland) [reply Lib/sqlite3/test/hooks.py:275]: Yep, see https://github.com/python/cpython/pull/92147#issuecomment-1115002300

Comment (erlend-aasland) [reply Lib/sqlite3/test/hooks.py:275]: So for this particular test, it is even harder; we have to _know_ the limit of SQLite strings, which we can only do by setting it to a known value. `sqlite3.Connection.setlimit` is not available in 3.10, so we'd have to rely on the compile-time constant the particular SQLite library was compiled with, or just aim for `INT_MAX`.

**Comment by JelleZijlstra** [comment]
> FTR, the 3.10 sqlite3 test suite lack a lot (all?) of the helper methods that are present in 3.11. We _could_ backport all those, but in the past, we've just removed the tests that were not applicable due to this limitation. I'm not sure backporting the test helpers are worth it; OTOH, its not a big deal to backport them.

Thanks for the explanation! I think it's helpful to keep test helpers as much as possible the same across branches, so all the branches are well-tested and backports are easy. But we don't have to do that right now.

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**